packages/addon-dev/src/rollup.ts

```diff
@@ -16,20 +17,28 @@ export class Addon {
   constructor(params: { srcDir?: string; destDir?: string } = {}) {
     this.#srcDir = params.srcDir ?? 'src';
     this.#destDir = params.destDir ?? 'dist';
+
+    let pkg = readJsonSync('package.json');
+    delete pkg['ember-addon']['app-js'];
```

Comment:
This scheme of only clearing `app-js` at startup and then forever merging it forward is not going to work correctly.

It means that deleted files will continue to exists in `app-js`, for example, which is a hard build error.